File: C#/0575. Distribute Candies.cs
public class Solution {
    public int DistributeCandies(int[] candies) {
        HashSet<int> candiesSet=new HashSet<int>();
        foreach(int candy in candies){
            candiesSet.Add(candy);
        }
        return candies.Length/2>candiesSet.Count()?candiesSet.Count():candies.Length/2;
    }
}
//====================================================================================================
public class Solution {
    public int DistributeCandies(int[] candies) {
        HashSet<int> candiesSet=new HashSet<int>(candies);
        return candies.Length/2>candiesSet.Count()?candiesSet.Count():candies.Length/2;
    }
}

File: 20190519python/0575. Distribute Candies.py
class Solution:
    def distributeCandies(self, candies: List[int]) -> int:
        dict_count={}
        for candy in candies:
            if candy not in dict_count:
                dict_count[candy]=1
            else:
                dict_count[candy]+=1
        
        return  len(candies)//2 if len(dict_count)>len(candies)//2 else len(dict_count)
